validator/src/validator.rs

```diff
@@ -342,6 +342,19 @@ where
                 epoch_number = blockchain.epoch_number(),
                 "We are ACTIVE in this epoch"
             );
+
+            // Find the entry for this validator in the epochs validators.
+            let epoch_validator = validators.get_validator_by_slot_band(slot_band);
+
+            // Compare configured validator voting key to the one in the current epoch to make sure it is the same.
+            if *epoch_validator.voting_key.compressed() != self.voting_key().public_key.compress() {
+                panic!("Invalid validator configuration: Configured voting key does not match voting key in this epoch");
+            }
+
+            // Compare configured validator signing key to the one in the current epoch to make sure it is the same.
+            if epoch_validator.signing_key != self.signing_key().public {
+                panic!("Invalid validator configuration: Configured signing key does not match signing key in this epoch");
+            }
```

Comment:
Is discussed in-person just now: the check below that checks against the staking contract can probably be removed.

Reply:
done